google/cloud/aiplatform/initializer.py

```diff
@@ -14,28 +14,142 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+import logging
+from typing import Dict, Optional
 
+import google.auth
+from google.auth import credentials as auth_credentials
+from google.cloud.aiplatform import utils
+
+
+def validate_region(region: str):
+    """Validates region against supported regions.
+
+    Args:
+        region: region to validate
+    Raises:
+        ValueError if region is not in supported regions.
+    """
+    region = region.lower()
+    if region not in SUPPORTED_REGIONS:
+        raise ValueError(
+            f"Unsupported region for AI Platform, select from {SUPPORTED_REGIONS}"
+        )
 
-from typing import Optional
 
 class Init:
+    """Stores common parameters and options for API calls."""
+
+    def __init__(self):
+        self._project = None
+        self._experiment = None
+        self._location = None
+        self._staging_bucket = None
+        self._credentials = None
+
+    def init(
+        self,
+        *,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        experiment: Optional[str] = None,
+        staging_bucket: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        """Updates common initalization parameters with provided options.
+
+        Args:
+            project: The default project to use when making API calls.
+            location: The default location to use when making API calls. If not set
+                defaults to us-central-1
+            experiment: The experiment to assign
+            staging_bucket: The default staging bucket to use to stage artifacts when
+                making API calls.
+            credentials: The default custom credentials to use when making API calls. If
+                not provided crendentials will be ascertained from the environment.
+        """
```

Comment:
Adding arg types to doc string

```suggestion
        """Updates common initialization parameters with provided options.

        Args:
            project (str): The default project to use when making API calls.
            location (str): The default location to use when making API calls. If not set
                defaults to us-central-1
            experiment (str): The experiment to assign
            staging_bucket (str): The default GCS staging bucket to use to stage artifacts
                when making API calls. For example, "gs://my-staging-bucket".
            credentials (google.auth.credentials.Credentials): The default custom 
                credentials to use when making API calls. If not provided credentials will
                be ascertained from the environment.
        """
```